Momento/Incubating/Responses/CacheSetFetchResponse.cs

```diff
@@ -0,0 +1,46 @@
+using System.Collections.Generic;
+using System.Linq;
+using Google.Protobuf;
+using Google.Protobuf.Collections;
+using CacheClient;
+using MomentoSdk.Internal;
+using MomentoSdk.Responses;
+
+namespace MomentoSdk.Incubating.Responses;
+
+public class CacheSetFetchResponse
+{
+    public CacheGetStatus Status { get; private set; }
+    private readonly RepeatedField<ByteString>? elements;
+
+    public CacheSetFetchResponse(_SetFetchResponse response)
+    {
+        Status = (response.SetCase == _SetFetchResponse.SetOneofCase.Found) ? CacheGetStatus.HIT : CacheGetStatus.MISS;
+        elements = (Status == CacheGetStatus.HIT) ? response.Found.Elements : null;
+    }
+
+    public HashSet<byte[]>? ByteArraySet
+    {
+        get
+        {
```

Comment:
this is pretty expensive. Don't you want to cache it in case of repeated calls?

Why is HashSet better than Set for this property's exposed type?

Reply:
I'll add in the caching. Didn't have an intuition as to the # of calls we would expect vs extra references in the class.

HashSet: in C# we have `ISet`, the set interface, and concrete types `HashSet`, `SortedSet`, and `Immutable{Hash,Sorted}Set`. In the code the pattern I've followed is: methods should accept arguments with the broadest interface possible (prefer `elements` as `IEnumerable<T>` to `ISet<T>`), but when we return a data structure, the concrete type is known, so specify that.

Reply:
2 calls seems like  a reasonable thing for a user to do:
```
if (2 < response.ByteArraySet.Count) {
  Console.WriteLine("a set of 3 or more: {0}", response.ByteArraySet);
}
```

and probably it's reasonable to call it K times in the general case. 1 reference in memory is worth reducing n * K to n.

Cool - maybe you should return ImmutableHashSet then?

Reply:
Added caching for `ByteArraySet` and `StringSet` in [004d5fb](https://github.com/momentohq/client-sdk-csharp/pull/93/commits/004d5fb5a70afbc1e0a2f5113dcf6069fe31bd68)